src/core/linter.js

```diff
@@ -0,0 +1,30 @@
+const path = require('path');
+const {parseCommentSuppression} = require(path.resolve(
+  __dirname,
+  '../utils/linterUtils.js',
+));
+
+class Linter {
+  constructor(rules, processor, parser) {
+    this.rules = rules;
+    this.parser = parser;
+    this.processor = processor;
+  }
+
+  verify(file, report, config) {
```

Comment:
One problem that I'm seeing is that we are iterating over and over the tokens,  but now, we can implement the pattern that you commented before `visitor`. We didn't implement previously because we were restricted by markdownlint API. I think this is very beneficial for us, because the creation of rules could be simpler, we can avoid repetitive code. But I'm not 100% if we should focus on this now 😆  waiting your thoughts 

[Another linter that uses visitor pattern](https://github.com/textlint/textlint/blob/07c0497b387f09e41088b7f113555aa5e5451add/packages/%40textlint/kernel/src/task/textlint-core-task.ts#L149), saving for inspiration, maybe

Reply:
it seems that performance is a no issue for now.

the visitor pattern may be something that we could provide for those who want it, but keep it side-by-side with the token evaluation that we use now.